### PR TITLE
Text input validation improvements

### PR DESCRIPTION
**Pull request by blammit**

### CODE REVIEW

**Comment by blammit** [comment]
I'm wondering whether `InputValidation_Result_Warning` confuses the API a bit, in terms of adding too many options. An alternative is to add an argument to `Utils.validationResult()` to specify the notification type, but I think it's preferable for ListTextField to automatically show the correct type of notification, than to add another configuration parameter.

**Review by DanielMcInnes — APPROVED** [review]
lgtm

**Comment by DanielMcInnes** [comment]
> I'm wondering whether `InputValidation_Result_Warning` confuses the API a bit, in terms of adding too many options. An alternative is to add an argument to `Utils.validationResult()` to specify the notification type, but I think it's preferable for ListTextField to automatically show the correct type of notification, than to add another configuration parameter.

I don't think adding `InputValidation_Result_Warning` makes the API confusing, it seems fine to me

File: components/Utils.js [hunk 1]
```diff
@@ -329,6 +329,7 @@ function toHexFormat(n) {
 // Returns an object to return for text field validation. The status can be:
 // - InputValidation_Result_Error: the input text is invalid; highlight the input with a red border
 // - InputValidation_Result_OK: the input text is valid, and can be saved, or red highlight can be removed
+// - InputValidation_Result_Warning: same as InputValidation_Result_OK, but show a warning (rather than info) notification
```
Comment (chriadam) [inline components/Utils.js:332]: this comment seems to suggest that InputValidation_Result_OK will cause an Info notification to be shown.  Is that the case?

Comment (chriadam) [reply components/Utils.js:332]: Ah, I see, that was added in 872b47990f04832976e817e822714de6f09f14d1.  Maybe the comment for InputValidation_Result_OK can be expanded to say: "An info toast notification will be raised to inform the user that validation succeeded" or something?

Comment (chriadam) [reply components/Utils.js:332]: Ah, but the OK was changed to Warning in this commit.  So, in the InputValidation_Result_OK case now, there should be no toast notifications raised, I believe.  If this is indeed the case, I would simply change this comment to read "but show a Warning-styled toast notification", perhaps.

**Review by chriadam — APPROVED** [review]
One nitpick, aside from that LGTM

File: components/Utils.js (hunk 1 above)
Comment (blammit) [reply components/Utils.js]: If `notificationText` is set, then a notification will still be shown, but it's optional. I've updated the docs to try to make it more clear about how  notifications are handled.

**Pull request merged**